app/api/auth.py

```diff
@@ -0,0 +1,182 @@
+from datetime import timedelta
+from app import db, jwt
+from flask import Blueprint, request, jsonify
+from flask_jwt_extended import create_access_token, jwt_required, get_jwt
+from app.models import Users, Account, TokenBlockList
+
+bp = Blueprint('auth', __name__)
+
+
+@bp.route('/signup', methods=['POST'])
+def signup():
+    try:
+        data = request.get_json()
+
+        attributes = data.get('data', {}).get('attributes', {})
+        username = attributes.get('username')
+        password = attributes.get('password')
+        account_name = attributes.get('account_name')
+
+        if not attributes or not username or not password or not account_name:
+            return jsonify({'errors': [{
+                'status': '422',
+                'detail': 'Some attributes are missing. '
+                'Username, password and account name are required'
+            }]}), 422
+
+        existing_user = Users.query.filter_by(username=username).first()
+        if existing_user:
+            return jsonify({'errors': [{
+                'status': '422', 
+                'detail': 'User already exists'
+            }]}), 422
+
+        existing_account = Account.query.filter_by(account_name=account_name).first()
+        if existing_account:
+            return jsonify({'errors': [{
+                'status': '422', 
+                'detail': 'Account name already exists'
+            }]}), 422
+
+        new_account = Account(account_name=account_name)
+        db.session.add(new_account)
+        db.session.commit()
+
+        new_user = Users(username=username, account_id=new_account.account_id, is_owner=True)
+        new_user.set_password(password)
+        db.session.add(new_user)
+        db.session.commit()
+
+        access_token = create_access_token(identity=new_user.username, expires_delta=timedelta(hours=2))
+
+        return jsonify({
+            'data': {
+                'id': new_user.user_id,
+                'type': 'users',
+                'attributes': {
+                    'message': 'Users has been successfully created',
+                    'token': access_token
+                },
+                'relationships': {
+                    'account': {
+                        'data': {
+                            'id': new_account.account_id,
+                            'type': 'accounts'
+                        }
+                    }
+                }
+            }
+        }), 201
+    except (KeyError, TypeError):
+        return jsonify({"errors": [{
+            "status": "400", 
+            "detail": "Invalid JSON structure"
+        }]}), 400
```

Comment:
Wrap in `try-except` only that part of the code, where you expect an error to occur.
If you wrap everything in `try-except` and make a programming error that handled in except, you won't notice it.
Always wrap in try-except as little as possible.